frontend/store/auctions.ts

```diff
@@ -147,25 +155,46 @@ export const mutations = {
 };
 
 export const actions = {
-    async update({ commit, rootGetters }: ActionContext<State, State>) {
+    async update({ rootState, commit, rootGetters }: ActionContext<State, any>, forceUpdateAllAuction?: boolean) {
         const network = rootGetters['network/getMakerNetwork'];
         if (!network) {
             return;
         }
-        commit('setAreAuctionsFetching', true);
-        try {
-            const auctions = await fetchAllAuctions(network);
-            const active = auctions.filter(auction => auction.isActive);
-            active.forEach(auction => {
-                commit('removeAuctionRestarting', auction.id);
-            });
-            commit('setError', null);
-            commit('setAuctions', auctions);
-        } catch (error) {
-            console.error('fetch auction error', error);
-            commit('setError', error.message);
-        } finally {
-            commit('setAreAuctionsFetching', false);
+        const selectedAuctionId = rootState.route.query.auction;
+        if (!selectedAuctionId || forceUpdateAllAuction) {
+            commit('setAreAuctionsFetching', true);
+            try {
+                const auctions = await fetchAllAuctions(network);
+                const active = auctions.filter(auction => auction.isActive);
+                active.forEach(auction => {
+                    commit('removeAuctionRestarting', auction.id);
+                });
+                commit('setError', null);
+                commit('setAuctions', auctions);
+            } catch (error) {
+                console.error('fetch auction error', error);
+                commit('setError', error.message);
+            } finally {
+                commit('refreshLastUpdated');
+                commit('setAreAuctionsFetching', false);
+            }
+        } else {
+            commit('setAreAuctionsFetching', true);
+            try {
+                const auctionInfo = parseAuctionId(selectedAuctionId);
+                const auction = await fetchAuctionByCollateralTypeAndAuctionIndex(
```

Comment:
Can you please instead create `fetchAuctionByAuctionId` which does parsing of the id inside of it?